Added SMM Firebar Block

KirbyKid256 opened the pull request:
First time adding a thing that could affect levels. I don't know what the rules are from that, but I attempt to keep it in relation to SMB1, All-Stars, and SMM. Probably better to merge this PR when you do a version update like from v1.0.1 to v1.0.2.

This adds an extra option for Firebars to toggle the collision and visibility of its block, as seen in Super Mario Maker.
It's off by default of course since most of the time it's attached to existing blocks, as well as to not screw with existing levels. But this makes use of an unused texture in the files.

<img width="407" height="366" alt="Screenshot 2025-09-19 111435" src="https://github.com/user-attachments/assets/f73011ac-4c32-4649-b18e-28c40c64efe4" />
<img width="1632" height="582" alt="Screenshot 2025-09-19 111420" src="https://github.com/user-attachments/assets/d3cdc2f6-5312-4722-a5fb-a39252bb2d04" />

KirbyKid256 committed:
Update Firebar.tscn

This adds an extra option for Firebars to toggle the collision and visibility of its block, as seen in Super Mario Maker.

It's off by default of course since most of the time it's attached to existing blocks. But this makes use of an unused texture in the files.

JeodC commented:
Just rebase.

KirbyKid256 committed:
Merge branch 'main' into pulls/firebar-block

LukaTV939 commented:
@KirbyKid256 could you make it so the firebar (with visible block) could move on tracks?

KirbyKid256 commented:
Is there a way to turn this into a draft PR so it doesn't get merged while I'm working on it?

LukaTV939 commented:
> Is there a way to turn this into a draft PR so it doesn't get merged while I'm working on it?

[yes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request)

KirbyKid256 committed:
Made Firebar Block Track connectable

Only when the block is visible though.

KirbyKid256 committed:
Update Firebar.tscn

KirbyKid256 commented:
https://github.com/user-attachments/assets/80d26d41-660a-41ed-8767-cd01c2d599fe

LukaTV939 commented:
> https://github.com/user-attachments/assets/80d26d41-660a-41ed-8767-cd01c2d599fe
> 

LFG, I can fully remake "1-1 but with a twist" now

KirbyKid256 committed:
Block visibility is toggled when changed in editor

KirbyKid256 commented:
Thanks 👍

LukaTV939 commented:
Found this completely randomly while making the level
When the firebar touches the track rider (with block on) it hops onto it

https://github.com/user-attachments/assets/7fbb3af0-da50-4241-a6ef-041dd574ce47

get's really messy if there's a lot of them track riders

KirbyKid256 committed:
Merge remote-tracking branch 'upstream/main' into pulls/firebar-block

KirbyKid256 committed:
Changed Fire Bar into a Block Class

Oh my god finally. This should help fix the issue with the bar touching the track rider causing the thing to attach.

KirbyKid256 commented:
https://github.com/user-attachments/assets/97f616c3-7489-4385-985f-c097aa7ad632

LukaTV939 commented:
just checked if it works properly, thank you
hope I don't run into more

LukaTV939 commented:
> > just checked if it works properly, thank you hope I don't run into more
> 
> Sike, found another one
> 
> This is how the firebar looks like:
> and if you play it, it snaps to 45°
> Here's a vid https://files.catbox.moe/tkp2r5.mp4

Ok now it's more apparent the more firebars there is
<details><summary>Here's the level so far:</summary>

```{"Info":{"Name":"1-1 BUT WITH A TWIST","Author":"YTSUNNY","Description":"","Difficulty":0},"Levels":[{"Layers":[{"-1":{"Tiles":"eJwly7kRACEMwMDKCIAK+A8yHrv/EmB0yUZSEGetieKfCTMWrNjww44DJy48KKi4/169ey82rFgwY8KIAQdOXLjxoKDiBTM5OFg%","Entities":"eJwLTdFxtNBxtAUACdwCCQ%%"},"0":{"Tiles":"eJwlzk0SgjAMhuGTsSgFhSogoFLZ9S/3P4LNy+aZTCb5ElOsMU2LHfZ4wzsOOKLDB1o0YtuagBY77PGGdxxwRIcPsdY0z0vSJpxxwReuuOGOb/zgFw/0V03mlysffOOOG674wgVnnMRONYeEOTn9Ko1qHnUmu+qPKz/p6+TJVsCICTMWlEu2CmZMGDHgef3PLc8tn9QjaeeI+omP+lVhptAvUZWok8K8sFsC/aB1ytZ0TcxtNWRTLaemyalpf66Sl30%","Entities":"eJzzTNRxTNZRVNRx0nG0dUfmuCJz3ICcJB3HIjDHA8Ixg+gJR+I4osik6Dg62QIAdVQXhg%%"},"1":{"Tiles":"eJwl0DkSwyAMQNGTpWBJbLAd7zMxHev9jxD4NK+QCv2RKEqKl0SFGt/46WYl6gQ1KpQocMARDVqccMYFv7jihjseeOKFd9bVX5bNImvJXXT1ourEA3fccMUvLjjjhBYNjjj0eTLNiKE5BduMmNA3rW/bh0KHHgNGTJixdLmSMWHEgB4dPr3HmfY9Z6t/N6J4fQ%%","Entities":"eJxLSdRxTNZRVNRx0nG0TUPmpAI5STqOpUCOky0Aw40JyQ%%"},"2":{"Tiles":"eJwlzjcOg0AURdGVuSC5AJNMpiMM+1+C8aE5unpfI00Ukih6xUyYPn0l8b0zYsrMNdPvK/0b0nup7XmIbwv9YcnqWbwqmF/xbakr1mzY8suOPQeOnDhz4cqNOw+eDLwe/Sfw5MGdG1cunDlx5MCeHb9s2fAHwOVkSQ%%","Entities":"eJxVzCkOgFAMBNCzfD8a90XLGsKiAF3gCoTrU6gZ5MssukEOpASF5JJRMWpGw2gZHWNgjIyJMTt2SBGwN3H0jsWotkZyfzCjzflDvF0OzQ85OTSv"},"3":{"Tiles":"eJwl0EkOglAQBNCTufigDMo8CLIDkfsfweT9zUula1PpcKVJuCVMo780hNs9Gls58OGSMWfBkk++WLFmw5Ydew4cOXHmmwtXfrhx58EvT/6idp788uDOjR+uXPjmzIkVX3yyZBb/wJwFaw4c2bBlx55X1No/uhxuaQ%%","Entities":"eJxNjTsOgDAMQ8/SPTNbh7AgAQsfqbRbG+D+R4A6ixfLz7YSraJNdJBRNE4/mIQAmBPBwrAybAwnw87XDoY90dODId2iGi/rmlvXUqHukRdssi/dIzcsDUvzBO0NfdyjfdC+nsB/DPBDBg%%"},"4":{"Tiles":"eJwl0DkSgzAQRNGTOdDCIoodYwMZCOn+R2D4k7zq6qBVI5OcMR+LbbKiI3sssMQKawzJix25xwFHnHDGBb+4qtlZaXDBGSccccAeO2yzFUP2Yk1TYYkFenRo0eCfdzfc8cATL4x4Y8KsPQsXnnjgjpvu40/VHXLCrPfG8N4em/c37iDNejfiA3Y6bqM%","Entities":"eJxNizsOgCAQBc9Cv7UdBSYKJKC1doD3P4PsbEMzed/jkxD8OZSxK1NTZiNJps0sE4z2Mk2bjOwL32KavLCs8IK3adpqZP/M7xDnZJfg39W0aeZkw/S1+QHmajLB"},"5":{"Tiles":"eJwl0EkOggAQBMCXeWCTTVZF0Rvr/58gFJc6dDqTSQdbFAS3kBFjJrwzZcacBUs+WLFmw5Ydez754sA3P5d7FB4JB774ZM+ODWtWfLBkwZwtM6a8M2HMiCGDK1/yc4GlONdY87O/FoejP7/8ceLMlRt3LlffzZE/Tpy5cOXGnX/ZSnUt","Entities":"eJwlzjkWgzAMRdENsQUXNAzW0BrSJaEn+6+C3m/u0ZEs6y/XNM9t/Zb7u9w+ZX+VdpZ+lDHKlHSSafIy2U12k9+SnwMdDTvuuOm6amXAjoaOoamySS6aJINLUoVUcvrBy2DX0VTTH2QYz5W7XU/9a385kEsc"},"6":{"Tiles":"eJwlz7kRg0AAA8DKHBwcmP9/nUP/JRivk400IylcaQivhCkjM+YMd5o8GaaMzJjzLVmwZMWaDVt27Dlw5MSZC1du3Hnw5OeKP+/4tJ82HNy5ceXCmRNHDuzZsWXDmhVLFv+n/AIYiFjh","Entities":""}},{"-1":{"Tiles":"","Entities":"eJwLjNRx8tNxdtZxBEHboCgQ1wUMoEJhoSAhNzdXVyDXCSgQEQQRgHLTUE0AADPiE88%"},"0":{"Tiles":"","Entities":"eJzzSNZx8tNxdtZxBEFb7xAQ180NyvWPgHBBwNXVxQUqHBgKEnZ1dXZ2dHR313ECQkfbyDSQoIsLWBVIICkAphkqAACwGRql"},"1":{"Tiles":"","Entities":"eJxzT9Nx8tPx8NBxBEHbiHAQ1wkIoAIAhEgHzQ%%"},"2":{"Tiles":"","Entities":"eJzzCdBx8tNxcnJ01AFDWwAl6wQP"},"3":{"Tiles":"","Entities":"eJzzjdJx8tNxc9NxBEFbvxAQ1wkMdEDQ0RYAkyQIPA%%"},"4":{"Tiles":"","Entities":"eJxzCtZx8tNxAwJXVzc3HUcdJx1HW69QkKCTi5OLi5sLSBAIbYOSQYKOjlBuUiRYjROUCwCODxGq"},"5":{"Tiles":"","Entities":"eJzz9ddx8tNxctJxBEHb8HAQ1w0IoAIhURB5EIAKAQBFgQxl"}},{"-1":{"Tiles":"","Entities":"eJwLjtJx8tdx0nG0DQuDsSKCISwn26AIGAsA0LMKBQ%%"},"0":{"Tiles":"","Entities":"eJwLStNx8tdx0nG0DQqBsULCYCzvUBjLMwnCcrJ1jICxkgJBLGegLAByhRF2"},"1":{"Tiles":"","Entities":"eJyLCNNx8tdx1nG0BQAOswKF"},"2":{"Tiles":"","Entities":"eJzz8ddx8tdx1nG0BQAN/AJy"},"3":{"Tiles":"","Entities":"eJzzTdRx8tdx1nG09QuEsQA18gT6"},"4":{"Tiles":"","Entities":"eJwLTNJx8tdx1nG0dY6AsbxDYKwkuBgAz4gKAA%%"},"5":{"Tiles":"","Entities":"eJzz9dNx8tdx1nG0DQmFscIjYCwAdQQHdg%%"}},{"-1":{"Tiles":"","Entities":"eJwLjtJxzNYJ1nEL1XHScbINioByvXUcgdywMAjXxRMsGxEM4Tr5gGUTU6GK3cDciHAI1zkJzE2BymZBeIkQnmsmhAs1GGJNElSnSzLYGgBBnCa4"},"0":{"Tiles":"","Entities":"eJxzjNBxzNYJ1nHW1nHScbJ1CoVyfXUcgVzPJCjXACzrlQLhuhqBZb2hil2LwbLuUK6bG1jWLwTCdQTz/JOhJoWDuQADzx38"}},{}],"Data":"A=A=O=A=A=GQ=GQ=","BG":"A=A=A=A=A=A=A="},{},{},{},{}],"Version":"1.0.2\r\n"}
```

KirbyKid256 commented:
So, I remade what you showed in the first video, but with a Goomba and Goombrat. This branch **doesn't** have the Firebar Block feature. Same thing happens with them. Implying this isn't a bug with the Firebar in particular and is a general issue with tracks.
You should try it yourself and make a Bug Report, but this will have to be a separate PR. I learned that's how that works the hard way.

I can't do the PR myself right now since I'm working on a massive one with multiplayer support.

https://github.com/user-attachments/assets/1fceafba-8bb3-4efd-a515-d2c8cb29e1cf

Lucoshi commented:
@JHDev2006 In my view, mixing SMB1, All-Stars, Deluxe, and SMM it's all good idea, so I really liked the gizmos as a novelty thing. So including this would be great!
